dsl/OpusEnum.cc

```diff
@@ -38,34 +40,104 @@ bool isOpusEnum(core::MutableContext ctx, ast::ClassDef *klass) {
     return id->symbol == core::Symbols::root();
 }
 
-void OpusEnum::patchDSL(core::MutableContext ctx, ast::ClassDef *klass) {
-    if (ctx.state.runningUnderAutogen) {
-        // TODO(jez) Verify whether this DSL pass is safe to run in for autogen
-        return;
+vector<unique_ptr<ast::Expression>> processStat(core::MutableContext ctx, ast::ClassDef *klass, ast::Expression *stat) {
+    auto *asgn = ast::cast_tree<ast::Assign>(stat);
+    if (asgn == nullptr) {
+        return {};
     }
 
-    if (!isOpusEnum(ctx, klass)) {
-        return;
+    auto *lhs = ast::cast_tree<ast::UnresolvedConstantLit>(asgn->lhs.get());
+    if (lhs == nullptr) {
+        return {};
+    }
+
+    auto *rhs = ast::cast_tree<ast::Send>(asgn->rhs.get());
+    if (rhs == nullptr) {
+        return {};
+    }
+
+    if (rhs->fun != core::Names::new_() && rhs->fun != core::Names::let()) {
+        return {};
+    }
+
+    if (rhs->fun == core::Names::new_() && !rhs->recv->isSelfReference()) {
+        return {};
     }
 
-    for (auto &stat : klass->rhs) {
-        auto *asgn = ast::cast_tree<ast::Assign>(stat.get());
-        if (asgn == nullptr) {
-            continue;
+    if (rhs->fun == core::Names::let()) {
+        auto recv = ast::cast_tree<ast::UnresolvedConstantLit>(rhs->recv.get());
+        if (recv == nullptr) {
+            return {};
         }
 
-        auto *rhs = ast::cast_tree<ast::Send>(asgn->rhs.get());
-        if (rhs == nullptr) {
-            continue;
+        if (rhs->args.size() != 2) {
+            return {};
         }
 
-        if (!(rhs->fun == core::Names::new_() && rhs->recv->isSelfReference())) {
-            continue;
+        auto arg0 = ast::cast_tree<ast::Send>(rhs->args[0].get());
+        if (arg0 == nullptr) {
+            return {};
         }
 
-        auto loc = asgn->rhs->loc;
-        asgn->rhs = ast::MK::Send2(loc, ast::MK::T(loc), core::Names::let(), move(asgn->rhs), ast::MK::Self(loc));
+        if (!(arg0->fun == core::Names::new_() && arg0->recv->isSelfReference())) {
+            return {};
+        }
     }
+
```

Comment:
There's a ton of nested pattern-matching happening above, could you put a short comment that describes what you expect the current expression to look like at this point?